chirp/drivers/radtel_rt493.py

```diff
@@ -0,0 +1,1152 @@
+# Copyright 2025 Paul Oberosler <paul@paulober.dev>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import time
+import struct
+import logging
+from chirp import bitwise, chirp_common, directory, errors, memmap, util
+from chirp.settings import (
+    RadioSetting,
+    RadioSettings,
+    RadioSettingGroup,
+    RadioSettingValueBoolean,
+    RadioSettingValueList,
+    RadioSettingValueInteger,
+    RadioSettingValueString
+)
+
+
+LOG = logging.getLogger(__name__)
+
+COMMAND_ACCEPT = b"\x06"
+CHARSET_HEX = "0123456789ABCDEFabcdef"
+
+
+def _clean_buffer(radio):
+    radio.pipe.reset_input_buffer()
+    radio.pipe.reset_output_buffer()
+    time.sleep(0.002)
+
+    radio.pipe.timeout = 0.005
+    junk = radio.pipe.read(256)
+    radio.pipe.timeout = 1  # 1000ms
+
+    if junk:
+        LOG.debug("Got %i bytes of junk before starting" % len(junk))
+
+
+def _rawrecv(radio, amount):
+    """Raw read from the radio device"""
+    data = bytearray()
+    try:
+        while len(data) < amount:
+            chunk = radio.pipe.read(amount - len(data))
+            if not chunk:
+                LOG.debug("No response from radio")
+                raise errors.RadioNoContactLikelyK1()
+            data.extend(chunk)
+            time.sleep(0.002)  # 2ms delay
+    except Exception:
+        msg = "Generic error reading data from radio; \
+            restart radio and check your cable."
+        raise errors.RadioError(msg)
+
+    if len(data) != amount:
+        LOG.error("Expected %i bytes, got %i" % (amount, len(data)))
+        raise errors.RadioError("Invalid response from the radio")
+
+    return bytes(data)
+
+
+def _rawsend(radio, data):
+    """Raw send to the radio device"""
+    try:
+        radio.pipe.write(data)
+        time.sleep(0.002)
+    except Exception:
+        raise errors.RadioError("Error sending data to radio")
+
+
+def _recv(radio, addr, length):
+    """Get data from the radio """
+    hdr = _rawrecv(radio, 4)  # Read 4 byte header
+    data = _rawrecv(radio, length)  # Read data
+
+    # DEBUG
+    LOG.info("Response:")
+    LOG.debug(util.hexprint(hdr + data))
+
+    c, a, resp_length = struct.unpack(">BHB", hdr)
+    if a != int.from_bytes(addr, byteorder="big") \
```

Comment:
Why are you doing this? If the byte order isn't actually big, use the little-endian `struct.unpack` format so you can compare directly.

Reply:
I'm not sure what you mean. I convert the addr to int with byteorder big because it is of type bytes.